service-proxy.js

```diff
@@ -95,10 +96,20 @@ function ServiceDispatchHandler(options) {
     self.periodicStatsTimer = null;
     self.statsPeriod = options.statsPeriod || DEFAULT_STATS_PERIOD;
 
+    self.peersToReap = Object.create(null);
+    self.connectedPeers = Object.create(null);
+    self.reapPeersTimer = null;
+    self.reapPeersPeriod = options.reapPeersPeriod || DEFAULT_REAP_PEERS_PERIOD;
```

Comment:
Nope, but it looks like we have quite similar names! I'll see myself out, carry on!;)